packages/node/example/multiSession/src/serverSideApp.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  getSessionFromStorage,
+  Session,
+  getSessionIdFromStorageAll,
+} from "../../../dist/Session";
+
+import cookieSession from "cookie-session";
+
+const clientApplicationName = "solid-client-authn-node multi session demo";
+
+import express from "express";
+const app = express();
+const PORT = 3001;
+
+const DEFAULT_OIDC_ISSUER = "https://broker.pod.inrupt.com/";
+
+app.use(
+  cookieSession({
+    name: "session",
+    // These keys are required by cookie-session to sign the cookies.
+    keys: ["key1", "key2"],
+    maxAge: 24 * 60 * 60 * 1000, // 24 hours
+  })
+);
+
+app.get("/", async (req, res, next) => {
+  const sessionIds = await getSessionIdFromStorageAll();
+  const sessions = await Promise.all(
+    sessionIds.map(async (sessionId) => {
+      return await getSessionFromStorage(sessionId);
+    })
+  );
+  const htmlSessions =
+    sessions.reduce((sessionList, session) => {
+      if (session?.info.isLoggedIn) {
+        return sessionList + `<li>${session?.info.webId}</li>`;
+      }
+      return sessionList + "<li>Anonymous</li>";
+    }, "<ul>") + "</ul>";
+  res.send(
+    `<p>There are currently ${sessionIds.length} visitors: ${htmlSessions}</p>`
+  );
+});
+
+app.get("/login", async (req, res, next) => {
+  const session = new Session();
+  req.session!.sessionId = session.info.sessionId;
+  await session.login({
+    redirectUrl: "http://localhost:3001/redirect",
+    oidcIssuer: DEFAULT_OIDC_ISSUER,
+    clientName: clientApplicationName,
+    handleRedirect: (redirectUrl) => res.redirect(redirectUrl),
+  });
+  if (session.info.isLoggedIn) {
+    res.send(`<p>Already logged in with WebID: [${session.info.webId}]</p>`);
+  }
+});
+
+app.get("/redirect", async (req, res) => {
+  const session = await getSessionFromStorage(req.session!.sessionId);
+  if (session === undefined) {
+    res
+      .status(400)
+      .send(`<p>No session stored for ID ${req.session!.sessionId}</p>`);
+  } else {
+    await session.handleIncomingRedirect(getRequestFullUrl(req));
+    if (session.info.isLoggedIn) {
+      res.send(`<p>Logged in as [${session.info.webId}] after redirect</p>`);
+    } else {
+      res.status(400).send(`<p>Not logged in after redirect</p>`);
+    }
+  }
+  res.end();
+});
+
+app.get("/fetch", async (req, res, next) => {
+  const session = await getSessionFromStorage(req.session!.sessionId);
+  if (!req.query["resource"]) {
+    res.status(400).send("<p>Expected a 'resource' query param</p>");
+  } else {
+    const fetch = (session ?? new Session()).fetch;
+    res.send(
+      "<pre>" +
+        (await (await fetch(req.query["resource"] as string)).text()).replace(
+          /</g,
+          "&lt;"
+        ) +
+        "</pre>"
+    );
+  }
+});
+
+app.get("/logout", async (req, res, next) => {
+  const session = await getSessionFromStorage(req.session!.sessionId);
+  if (session) {
+    session.logout();
+    res.send(`<p>Logged out</p>`);
```

Comment:
Again, just more info on logout (I assume the same WebID could login multiple times and have multiple sessions, so just to explain that so the developer doesn't think logging out of one of their session logs them out of all their sessions):
```suggestion
    const sessionInfo = session.info;
    session.logout();
    res.send(`<p>Logged out WebID [<strong>${sessionInfo.webId}</strong>] (which had session ID [<strong>${sessionInfo.sessionId}</strong>] (the same WebID could be logged into multiple sessions!))</p>`);
```

Reply:
837acf456a0fb34a401b067aca4e7d5206e6c340